noxfile.py

```diff
@@ -165,6 +165,11 @@ def tests(session):
 
     cache_cartopy(session)
     session.install("--no-deps", "--editable", ".")
+
+    # Always echo the env content for CI test failure debugging.
+    session.install("beautifultable")
+    session.run("python", "tools/env_listing.py")
```

Comment:
@pp-mo Personally, I'd opt to add the following 3 line `conda_script` in `.cirrus-ci` instead:
```python
#
# YAML alias for common linux test infra-structure.
#
linux_task_template: &LINUX_TASK_TEMPLATE
  auto_cancellation: true
  ...
  nox_cache:
     ...
  conda_script:
    - conda list --name base
    - conda list --name base --explicit
    - conda info
```